minari/dataset/minari_dataset.py

```diff
@@ -307,31 +255,16 @@ def update_dataset_from_collector_env(self, collector_env: DataCollectorV0):
             f"additional_data_{self._additional_data_id}.hdf5",
         )
 
-        collector_env.save_to_disk(path=new_data_file_path)
-
-        with h5py.File(new_data_file_path, "r", track_order=True) as new_data_file:
-            new_data_total_episodes = new_data_file.attrs["total_episodes"]
-            new_data_total_steps = new_data_file.attrs["total_steps"]
-
-        with h5py.File(self._data.data_path, "a", track_order=True) as file:
-            last_episode_id = file.attrs["total_episodes"]
-            for id in range(new_data_total_episodes):
-                file[f"episode_{last_episode_id + id}"] = h5py.ExternalLink(
-                    f"additional_data_{self._additional_data_id}.hdf5", f"/episode_{id}"
-                )
-                file[f"episode_{last_episode_id + id}"].attrs.modify(
-                    "id", last_episode_id + id
-                )
-
-            # Update metadata of minari dataset
-            file.attrs.modify(
-                "total_episodes", last_episode_id + new_data_total_episodes
-            )
-            file.attrs.modify(
-                "total_steps", file.attrs["total_steps"] + new_data_total_steps
-            )
+        self._data.update_from_collector_env(
+            collector_env, new_data_file_path, self._additional_data_id
+        )
+
         self._additional_data_id += 1
 
+        self._episode_indices = np.arange(self._data.total_episodes)
+
```

Comment:
It breaks filtered `MinariDataset`; you should append the new indices

Reply:
done